GitUI/CommandsDialogs/FormCreateBranch.cs

```diff
@@ -41,6 +41,19 @@ public FormCreateBranch(GitUICommands commands, GitRevision revision)
         public bool UserAbleToChangeRevision { get; set; }
         public bool CouldBeOrphan { get; set; }
 
+        protected override bool ProcessCmdKey(ref Message msg, Keys keyData)
+        {
+            // Handle Ctrl+Backspace combination in the text input (remove last word).
+            // More detail: https://stackoverflow.com/a/1197339/2009373
```

Comment:
What about the second alternative? 

Should this be applied globally?

Reply:
The global change looks quite dangerous to me (e.g. it could break complex `Message` field in the `Commit` dialog) and should be carefully tested. That's why I decided to proceed with a local change only, where form is tiny as it's easy to test it.

Reply:
I use <kbd>Ctrl</kbd>+<kbd>Backspace</kbd> all the time, and it bugs me when a text box doesn't support it. Thanks for this PR.

> Should this be applied globally?

I'd prefer a global fix to a local one.

If it requires overriding `ProcessCmdKey` then it'd be hard to do. I agree the second SO answer looks promising, as it [shows an alternative](https://superuser.com/a/332034/31787).

> The "delete word" behavior was added later, [as an undocumented feature](http://blogs.msdn.com/b/oldnewthing/archive/2007/10/11/5395501.aspx), and only to those Edit controls which use "SHAutoComplete". If a textbox has autocompletion disabled, <kbd>Ctrl</kbd><kbd>Backspace</kbd> will likely not work.

I tried this out in the method we use to walk the control tree to fix DPI issues, and it seems to work well:

```diff

diff --git a/GitExtUtils/GitUI/ControlDpiExtensions.cs b/GitExtUtils/GitUI/ControlDpiExtensions.cs
index af8541452..4d7d5bfdc 100644
--- a/GitExtUtils/GitUI/ControlDpiExtensions.cs
+++ b/GitExtUtils/GitUI/ControlDpiExtensions.cs
@@ -2,6 +2,7 @@
 using System.Collections.Generic;
 using System.ComponentModel;
 using System.Drawing;
+using System.Runtime.InteropServices;
 using System.Windows.Forms;
 using GitExtUtils.GitUI;
 using JetBrains.Annotations;
@@ -95,11 +96,18 @@ public static void AdjustForDpiScaling([NotNull] this Control control)
                         break;
                     }

-                    case TextBoxBase textBox when next.Margin == new Padding(12):
+                    case TextBoxBase textBoxBase when next.Margin == new Padding(12):
                     {
+                        if (textBoxBase is TextBox textBox)
+                        {
+                            const uint SHACF_AUTOSUGGEST_FORCE_OFF = 0x20000000;
+
+                            SHAutoComplete(textBox.Handle, SHACF_AUTOSUGGEST_FORCE_OFF);
+                        }
+
                         // Work around a bug in WinForms where the control's margin gets scaled beyond expectations
                         // see https://github.com/gitextensions/gitextensions/issues/5098
-                        textBox.Margin = DpiUtil.Scale(new Padding(3));
+                        textBoxBase.Margin = DpiUtil.Scale(new Padding(3));
                         EnqueueChildren();
                         break;
                     }
@@ -129,5 +137,8 @@ void EnqueueChildren()
                 }
             }
         }
+
+        [DllImport("shlwapi.dll", ExactSpelling = true, PreserveSig = false)]
+        private static extern int SHAutoComplete(IntPtr hwndEdit, uint flags);
     }
 }
\ No newline at end of file
```

This will only apply to `TextBox`. I ran some tests for a few minutes and it seems to work well across the board.

@zvirja what do you think of the approach outlines in the above diff? It might not apply cleanly to master as I based it off the `v3-ux` branch from #5087.

Reply:
Nice! Let's apply it globally, if possible.

Reply:
@drewnoakes Thanks for the insights! Probably it's indeed better to follow the approach you described.
Let's wait till #5087 is merged and proceed after that!

Could you please ping me when you are ready?

Reply:
We're all sitting and waiting for that moment  🍺 🍿 🕐

Reply:
@zvirja #5087 has been merged now so feel free to pick this up again and we'll get it merged in.